### PR TITLE
feat: Add API GTFS Schedule sources compatibility 

### LINKED ISSUE #130: Add API GTFS schedule sources compatibility
**What problem are we trying to solve?**

Right now, GTFS schedule sources that are served by an API cannot be included in the database. We want to be able to add them.

**How do we know when this is done?**

- The README is updated to include the same authentication fields for GTFS Schedule that currently exist for GTFS Realtime
- Update GitHub workflow so keys can be stored and used for sources with an authentication type greater than 0 

**Tasks**

- #185 
- #189 
- [ ] Update GTFS schedule schema to include authentication fields
- [ ] Update operations to include authentication info 
- [ ] Update latest dataset URL workflow

### PR DESCRIPTION
**Pull request by maximearmstrong**
**Summary:**

Closes #130: Add API GTFS Schedule sources compatibility 

This PR updates the GTFS Realtime operations, representation and schema so that API sources (with authentication) can be added to the catalogs.

Changes:

- `gtfs_schedule_source_schema.json` so the GTFS Schedule schema accepts API sources.
- Both `direct_download_urls_test_for_sources.yml` and `store_latest_datasets_cronjob.yml` workflows to support API sources.
- `representations.py`, `operations.py` and their tests are updated to support API sources for GTFS Schedule.
- The schema and the operation examples are updated in`README.md`.
- `API_KEY_PARAMETER_VALUE` has been added in `constants.py`.
- `download_dataset()` have been updated in `helpers.py` to support authentication fields. The tests have been updated.

**Expected behavior:** 

- Users can now add their API sources using the `add_gtfs_schedule_source` operation. If the source has `authentication_type = 1` or `authentication_type = 2`, users must provide a value to the operation parameter `api_key_parameter_value` which is their private key. This value is not stored in the source file and representation. 
- Users can update their sources similarly, using `update_gtfs_schedule_sources`.
- Workflows now support API sources. The secret will be a JSON string of the following format: `{"source_base_name": "api_key_parameter_value"}`, where `"source_base_name"` is the source file name less its extension. Note that the `"source_base_name"` could eventually be changed to `"mdb_source_id"`, but since the base name was already used in both workflow, it seemed to be the most logical solution for now.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `pytest` to make sure you didn't break anything
- [x] Format the title like "<short description of fix and changes>" (for example - "Check for null value before using field")
- [x] Linked all relevant issues
- [] ~Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Comment by emmambd** [comment]
`users must provide a value to the operation parameter api_key_parameter_value which is their private key.` @maximearmstrong can you clarify how this works in practice? I assumed the workflow would be 

1. User submits PR with `api_key_parameter_name` 
2. We have to generate our own private key via the `authentication_info`
3. Merge PR 

Whereas based on your above statement it sounds like we're storing their personal private key.

**Comment by maximearmstrong** [comment]
> @maximearmstrong can you clarify how this works in practice? I assumed the workflow would be
> 
> 1. User submits PR with `api_key_parameter_name`
> 2. We have to generate our own private key via the `authentication_info`
> 3. Merge PR
> 
> Whereas based on your above statement it sounds like we're storing their personal private key.

Oh, sure I can clarify. I previously added "This value is not stored in the source file and representation." to that paragraph, but we can change it if it's not clear enough. 

These steps are correct, but there would be another one before:

- User locally adds source with `api_key_parameter_value` through the `add_gtfs_schedule_source` operation. `api_key_parameter_value` is **not** stored in the source.

This allows to download the source, assert it's a GTFS Schedule one and extract the bounding box. Without the `api_key_parameter_value`, there is no way to download it locally.

File: README.md [hunk 1]
```diff
@@ -215,13 +218,17 @@ get_sources_by_status(
     )
 ```
 
-To add a new GTFS Schedule source:
+To add a new GTFS Schedule source.  Note that the value passed to the `api_key_parameter_value` parameter will not be stored, though it is needed to add your source information if your source has `authentication_type = 1` or `authentication_type = 2`:
```
Comment (emmambd) [inline README.md:221]:
```suggestion
To add a new GTFS Schedule source. Note that you must pass an `api_key_parameter_value` if your source has `authentication_type = 1` or `authentication_type = 2`. The `api_key_parameter_value` will not be stored and is used only for testing before the source is added to the database.
```

**Comment by emmambd** [comment]
@maximearmstrong Got it, that's clear! So in order to test it we ask them to share a value, but we aren't actually storing that value and using it in the database itself. 

I made a suggested change to the README so this is clear. I also think this documentation change should be added to `CONTRIBUTING.md`.

**Review by emmambd — APPROVED** [review]

**Pull request merged**